### DISCUSSION TITLE
Custom authentication does not work

### DISCUSSION THREAD

**Discussion by matheogirardvoy** [filamentphp/filament#8566] [Help]
Hello,

I'm using Filament V3.

I'm trying to integrate Socialite into Filament to do custom authentication.
After several tries, Filament does not want to connect me.

Own code : 
```PHP
class AzureController extends Controller
{
	public function callback()
	{
		$socialiteUser = Socialite::driver('azure')->user();

		$user = Us3er::where('email', $socialiteUser->getEmail())->first();

		if ($user === null) {
			// Create user
		}

		Filament::auth()->login($user, true);
		return redirect(Filament::getHomeUrl());
	}
}
```

After that, Filament redirect me to login page.

Any ideas ?

Thanks 🙏

**Comment by leandrocfe** [comment]
There was a plugin called [filament-socialite](https://github.com/DutchCodingCompany/filament-socialite/tree/support-filament-v3) for v2. I'm not sure if it has support for v3, but you can take a look.

**Reply by matheogirardvoy** [reply]
Yes indeed, only it is not compatible with V3.
But I don't understand why `Filament::auth()->login()` is not working.
